blocks/identity-block/components/social-sign-on/_children/FacebookSignIn.jsx

```diff
@@ -1,20 +1,53 @@
-import React from "react";
+import React, { useEffect } from "react";
+
+import { Button, Icon, usePhrases } from "@wpmedia/arc-themes-components";
 
 import { SIGN_UP } from "../constants";
 
-function FacebookSignIn({socialSignOnIn}) {
-	const facebookTextType = socialSignOnIn === SIGN_UP ? 'continue_with' : 'login_with';
+const FacebookIcon = <Icon name="Facebook" />;
+
+function FacebookSignIn({ customButtons, socialSignOnIn, className }) {
+	const phrases = usePhrases();
+	const facebookTextType = socialSignOnIn === SIGN_UP ? "continue_with" : "login_with";
+
+	useEffect(() => {
+		if (customButtons) {
+			document
+				.getElementById("facebook-btn")
+				.addEventListener("click", () => window.onFacebookSignOn());
+		}
+	}, [customButtons]);
```

Comment:
This useEffect that adds an event listener should also remove it when it's cleaned up.  The standard pattern here is:

```
useEffect(() => {
    const listener = () => {}
    element.addEventListener("event", listener)
    return () => element.removeEventListener("event", listener)
}, [dependency]);
```

That return can be inside the condition.

But in this case, you are finding the element that you have created yourself, so why don't you just add the click listener to the Button?  This way you don't need the useEffect.

```
			<Button
				id="facebook-btn"
				variant="secondary-reverse"
				iconLeft={FacebookIcon}
				className={`${className}__Facebook`}
				onClick={() => window?.onFacebookSignOn?.() }
			>
```